src/filestorage.rs

```diff
@@ -8,26 +8,26 @@ use std::fs::{self, File};
 use std::io;
 use std::path;
 use std::rc::Rc;
+use std::slice;
 
 /// Internal storage of data as files.
 #[derive(Debug, Default)]
 struct MemoryMappedFileStorage {
-    maps: BTreeMap<String, Mmap>,
+    maps: RefCell<BTreeMap<String, Mmap>>,
 }
 
 impl MemoryMappedFileStorage {
-    pub fn read(&mut self, path: &str) -> Result<MemoryDescriptor, io::Error> {
-        if let Some(mapping) = self.maps.get(path) {
-            return Ok(MemoryDescriptor::new(mapping.as_ptr(), mapping.len()));
+    pub fn read(&self, path: &str) -> Result<&[u8], io::Error> {
```

Comment:
I guess, you are using interiour mutability, since resource storage is used multiple times for reading and therefore a single mut ref does not work for us. Right?

Reply:
Reading would have been fine: I prototyped a 2-phase approach: First loading all files, then creating views from them in the archive. Works perfectly fine.
Fails when it comes to creating Writers since you cannot have multiple mutable references.
So overall I settled on interior mutability, since the internal state (what is open, what is not) is actually not observable from the outside.